test/gui/shared/steps/steps.py

```diff
@@ -332,3 +332,18 @@ def step(context, number):
     # It might take some time for all files to sync except the expected number of unsynced files
     snooze(10)
     clickTab(waitForObject(names.stack_QTabWidget), "Not Synced ({})".format(number))
+
+@When('the user opens the public links dialog of "|any|" using the client-UI')
+def step(context, resource):
+    resource = substituteInLineCodes(context, resource).replace('//','/')
+    waitFor(lambda: isFileSynced(resource), context.userData['clientSyncTimeout'] * 1000)
+    waitFor(lambda: shareResource(resource), context.userData['clientSyncTimeout'] * 1000)
+    mouseClick(waitForObject(names.qt_tabwidget_tabbar_Public_Links_TabItem), 0, 0, Qt.NoModifier, Qt.LeftButton)
+
+@When("the user toggles the password protection using the client-UI")
+def step(context):
+    clickButton(waitForObject(names.oCC_ShareLinkWidget_checkBox_password_QCheckBox))
+
+@Then('the progress indicator should not be visible in the client-UI')
+def step(context):
+    test.compare(waitForObjectExists(names.oCC_ShareLinkWidget_checkBox_password_QProgressIndicator).visible, False)
```

Comment:
does s.th. like this work in Python?
```suggestion
    test.compare(
         waitForObjectExists(names.oCC_ShareLinkWidget_checkBox_password_QProgressIndicator).visible,
         False
    )
```

Reply:
It's not giving me any error so I think it works.